src/Cryptol/TypeCheck/Solver/Numeric.hs

```diff
@@ -223,6 +237,33 @@ tryCancelVar ctxt p t1 t2 =
 
 
 
```

Comment:
I think instead of the `tryEqMulExp`, we should add a rewrite to types, to turn `x^a * x ^b` into `x ^ (a + b)`.  Then `tryEqExp` should handle this rule.

Reply:
Updated in [110a753](https://github.com/GaloisInc/cryptol/pull/1713/commits/110a753a94506829ac11402ccf4f0067bb18bebe)